KAN-256-로그인회원가입구현

kimgorok opened the pull request:
## 😀 제목 : 로그인과 회원가입 실제 api 연결

<br/>

## 🔎 작업 내용

- 어떤 변경 사항이 있었나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

<br/>

## 🎨 이미지 첨부

<br/>

## 🔧 앞으로의 과제

- 다음으로 할 일은 무엇인가요??
유저 로그인 후 JWT 토큰을 쿠키에 저장합니다. 
=> (12.05) accessToken을 쿠키에 저장하고, 유저 정보를 불러오는 훅을 작성했습니다.
SNS 로그인을 구현합니다.

  <br/>

## ➕ 추가 코멘트
- (12.03)
1. 회원가입 함수를 연결하고 유효성 검사 규칙을 수정하였습니다.

- (12.04)
1. 로그인 시 발급되는 JWT 토큰을 브라우저 쿠키에 저장하도록 하였습니다.

- (12.05)
1. 로그인한 유저의 정보를 불러오는 useMe훅을 작성하였습니다.

<br/>

## 🤔 Issue Ticket

- [Ticket](https://gwagseoyeon.atlassian.net/browse/KAN-256)

kimgorok committed:
✨feat: [블로그] 직군 불러오기 및 post 함수 수정

- 타입 추가 및 위치 변경
- 기본 값으로 선택한 직군 저장
- 새 직군 선택 시 쿼리 무효화 적용

KAN-246

kimgorok committed:
✨feat: [블로그] 현재 선택 직군 검사

- 선택한 직군이 있으면 직군 선택 모달 안나오게 수정

KAN-246

kimgorok committed:
🐛fix: [블로그] 이미지 표시 오류 수정

- 이미지가 'null'일 경우 기본 이미지 표시

KAN-246

kimgorok committed:
🎉init: [블로그] 포스트 검색 페이지 및 함수 세팅

- /main/search 페이지 작성
- fetchPostSearch 함수 작성
- 검색 기능 구현

KAN-246

kimgorok committed:
✨feat: [블로그] 검색 페이지 클린업 함수와 스켈레톤 UI 추가

- 컴포넌트 리렌더링 시 클린업 수행
- 스켈레톤 UI 적용

KAN-246

kimgorok committed:
✨feat: [블로그] 검색 버튼 함수 연결 및 반응형 UI 적용

- Icons의 onClick에 handleSearch함수 연결
- 반응형 UI 적용

KAN-246

kimgorok committed:
✨feat: [블로그] 헤더에서 검색 페이지 이동 추가

- 헤더의 SearchIcon에 라우터 이동 연결

KAN-246

kimgorok committed:
💄design: [블로그] 검색 페이지 UI 수정

- 돋보기 아이콘에 cursor-pointer 추가

KAN-246

kimgorok committed:
✨feat: [블로그] content에서 검색어가 포함된 부분 표시

- post.content에서 검색어와 일치하는 부분 노란색으로 표시

KAN-246

kimgorok committed:
✨feat: [블로그] 검색어의 특수문자 이스케이프 처리

- replace 활용

KAN-246

kimgorok committed:
♻️refactor: [블로그] 메인/검색 페이지 레이아웃 정리

- /main/layout.tsx 작성

KAN-246

kimgorok committed:
🐛fix: [블로그] 타입 수정

- 타입 축소 및 위치 수정

KAN-246

kimgorok committed:
✨feat: [블로그] 유저 포스트 조회에 무한스크롤 적용

- useCustomInfiniteQuery를 사용해 기존과 같이 구현

KAN-246

kimgorok committed:
✨feat: 로그인/회원정보 조회

- 로그인 함수 postLogin 구현
- 로그인한 회원 정보 조회 추가

KAN-246

kimgorok committed:
🎉init: [회원가입] 회원가입 함수 구현

- 중복확인/회원가입 함수 작성
- 아이디/이메일 중복검사 로직 구현
- 폼 유효성 검사 변수 수정
- 타입 수정

KAN-256

kimgorok committed:
✨feat: [회원가입] 회원가입 구현

- SignUpForm 유효성 검사 및 회원가입 함수 구현 완료

KAN-256

kimgorok committed:
✨feat: [로그인] 로그인 시 쿠키에 JWT 토큰 정보 추가

- js-cookie라이브러리 설치
- 로그인 성공 시 쿠키 저장
- axios 인스턴스를 통해 헤더에 인증 토큰 전달

KAN-256

kimgorok committed:
✨feat: blogServie api 주소 수정

- 타인 정보 조회 fetchProfile 주소 수정

KAN-256

kimgorok committed:
🐛fix: [로그인/회원가입] 브라우저 history에 남게 수정

- router.push를 router.repace로 변경

KAN-256

kimgorok committed:
🐛fix: 회원 정보 불러오기 전 기타 수정사항

- fetchMemberUsername 삭제(중복 작성)
- 관심직군 선택 모달 표시 훅 수정
- 포스트 목록 lastId 수정

KAN-256

kimgorok committed:
🎉init: 로그인 유저 정보 불러오는 훅 작성

- useMe 작성 후 캐싱 처리

KAN-256

yeonilil commented on src/app/(route)/(blog)/blog/[username]/_components/PostContainer.tsx:
```diff
@@ -3,34 +3,60 @@
+  }, [inView, hasNextPage, isFetchingNextPage, fetchNextPage]);
 
-  const { loveMutation, bookmarkMutation } = useLoveAndBookmark(posts, setPosts, memberId, lastId);
+  const { loveMutation, bookmarkMutation } = useLoveAndBookmark(
```
아 love가 뭔가 했더니 공감 버튼인가요? ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

kimgorok committed:
🎉init: 카카오 로그인 세팅

- 로그인 정보 저장까지 구현

KAN-256

kimgorok committed:
✨feat: 헤더에 현재 로그인한 회원 정보 추가

- 헤더에 useMe 적용

KAN-256

kimgorok committed:
🎉init: [회원가입] 추가 정보 입력 페이지 제작

- SignUpDetailForm.tsx 페이지 제작
- 아이디 최소 6자 이상 validation 추가
- postSignUpDetail 함수 제작

KAN-256

kimgorok committed:
🐛fix: [로그인] 로그인폼 수정

- 로그인 시 기존 캐시 무효화
- value가 있을 때에만 value를 포함하는 inputProps 객체 생성

KAN-256

kimgorok committed:
🐛fix: accessToken이 있는 상태에서 로그인 페이지 접속 시, main으로 리다이렉트

- 로그인/회원가입 공통 layout.tsx에 accessToken 검사 로직 추가

KAN-256

kimgorok committed:
✨feat: 헤더 로그인 여부에 따라 조건부 렌더링

- 헤더에 조건부 렌더링 적용
- 로그아웃 함수 추가
- UI 수정

KAN-256

yeonilil commented on src/app/(route)/(log-in_sign-up)/layout.tsx:
```diff
@@ -0,0 +1,13 @@
+
+export default function RootLayout({ children }: { children: React.ReactNode }) {
+  const cookieStore = cookies(); // 서버 컴포넌트에서 쿠키를 읽기 위해 next/headers의 cookies() 사용
+  const accessToken = cookieStore.get('accessToken');
```
액세스 토큰으로 처리하셨군요! 좋습니당 👍

yeonilil commented on src/app/(route)/(log-in_sign-up)/layout.tsx:
```diff
@@ -0,0 +1,13 @@
+
+  if (accessToken) {
+    redirect('/main'); // redirect()는 서버 사이드에서 동작하며, 페이지 렌더링 전에 실행됨
+  }
```
여기에 redirect를 넣으신 이유가 뭔가요?

kimgorok replied:
사용자가 로그인을 한 상태에서(브라우저에 accessTokend이 있을 때) /log-in 페이지나 그와 관련된 페이지에 진입했을 때 /main으로 보내기 위함입니다.

BaekJiyeon02 commented on src/app/(route)/(log-in_sign-up)/sign-up/_components/SignUpForm.tsx:
```diff
@@ -2,186 +2,194 @@
-  const isFormValid = Object.keys(errors).length === 0;
+  const isFormValid =
+    isValid &&
+    isDirty && // 폼이 한번이라도 수정 되었는지
```
폼이 한번이라도 수정 되었는지 여부를 확인하는 이유가 궁금해요!

kimgorok replied:
사용자가 유효성검사를 통과한 상태에서, 최종 회원가입 버튼을 누르지 않고, 다시 폼을 수정하였을 때 최종 회원가입 버튼을 비활성화 하기 위함입니다

yeonilil commented on src/app/_components/Header/_components/SideMenu.tsx:
```diff
@@ -4,41 +4,42 @@ import { AnimatePresence, motion } from 'framer-motion';
-              <div className="pl-6 font-semibold text-xl">
-                <Link href="/my-page/check-password">{testData.userData.userName} 님</Link>
-                <div className="flex bg-gray-5 w-36 h-[1px] mt-3" />
+              <div className="font-semibold text-xl border-b border-gray-5 pb-3 pl-6 mr-6">
```
로그인 여부에 따라 처리해주신 부분 좋습니당

BaekJiyeon02 reviewed (approved):
로그인 회원가입과 Header까지 수정해주시다니.. 정말 고생많으셨습니다!!!
